L1Trigger/CSCTriggerPrimitives/src/CSCGEMMotherboard.cc

```diff
@@ -90,430 +104,397 @@ void CSCGEMMotherboard::run(const CSCWireDigiCollection* wiredc,
+          selectedLCTs.push_back(LCTsecondCLCTgem);
+        else if (LCTbestAsecondC.isValid())
+          selectedLCTs.push_back(LCTbestAsecondC);
+      } else if (lctbb.getQuality() < lctbs.getQuality() and lctbs.isValid()) {
+        selectedLCTs.push_back(lctbs);
+        if (LCTbestCLCTgem.isValid() and build_lct_from_alct_gem_)
+          selectedLCTs.push_back(LCTbestCLCTgem);
+        else if (LCTbestAbestC.isValid())
+          selectedLCTs.push_back(LCTbestAbestC);
+      }
 
-  // at this point, we have at least one valid cluster, and we're either dealing
-  // wit GE1/1 or GE2/1 with 16 partitions, both are OK for GEM-CSC trigger
+      sortLCTs(selectedLCTs);
 
-  // before matching ALCT-CLCT pairs with clusters, we check if we need
-  // to drop particular low quality ALCTs or CLCTs without matching clusters
-  // drop low quality CLCTs if no clusters and flags are set
-  GEMInternalCluster bestALCTCluster, secondALCTCluster;
-  GEMInternalCluster bestCLCTCluster, secondCLCTCluster;
-  cscGEMMatcher_->bestClusterBXLoc(bestALCT, clusters, bestALCTCluster);
-  cscGEMMatcher_->bestClusterBXLoc(secondALCT, clusters, secondALCTCluster);
-  cscGEMMatcher_->bestClusterBXLoc(bestCLCT, clusters, bestCLCTCluster);
-  cscGEMMatcher_->bestClusterBXLoc(secondCLCT, clusters, secondCLCTCluster);
+      for (unsigned iLCT = 0; iLCT < std::min(unsigned(selectedLCTs.size()), unsigned(CSCConstants::MAX_LCTS_PER_CSC));
+           iLCT++) {
+        if (selectedLCTs[iLCT].isValid()) {
+          selectedLCTs[iLCT].setTrknmb(iLCT + 1);
+          allLCTs_(bx_alct, matchingBX, iLCT) = selectedLCTs[iLCT];
+        }
+      }
+    }
 
-  dropLowQualityALCTNoClusters(bestALCT, bestALCTCluster);
-  dropLowQualityALCTNoClusters(secondALCT, secondALCTCluster);
-  dropLowQualityCLCTNoClusters(bestCLCT, bestCLCTCluster);
-  dropLowQualityCLCTNoClusters(secondCLCT, secondCLCTCluster);
+    // CASE => bestALCT, secondALCT, bestCLCT, secondCLCT are valid
+    if (bestALCT.isValid() and secondALCT.isValid() and bestCLCT.isValid() and secondCLCT.isValid()) {
+      CSCCorrelatedLCTDigi lctbb, lctbs, lctsb, lctss;
+      if (LCTbestAbestCgem.isValid())
+        lctbb = LCTbestAbestCgem;
+      else if (LCTbestAbestC.isValid())
+        lctbb = LCTbestAbestC;
+      if (LCTbestAsecondCgem.isValid())
+        lctbs = LCTbestAsecondCgem;
+      else if (LCTbestAsecondC.isValid())
+        lctbs = LCTbestAsecondC;
+      if (LCTsecondAbestCgem.isValid())
+        lctsb = LCTsecondAbestCgem;
+      else if (LCTsecondAbestC.isValid())
+        lctsb = LCTsecondAbestC;
+      if (LCTsecondAsecondCgem.isValid())
+        lctss = LCTsecondAsecondCgem;
+      else if (LCTsecondAsecondC.isValid())
+        lctss = LCTsecondAsecondC;
+
+      if (lctbb.getQuality() >= lctbs.getQuality()) {
+        selectedLCTs.push_back(lctbb);
+        selectedLCTs.push_back(lctss);
+      } else if (lctbb.getQuality() < lctbs.getQuality()) {
```

Comment:
```suggestion
      } else {
```
and then `lctsb` could be more conveniently computed here inside, rather than uselessly here above